src/testdir/test_edit.vim

```diff
@@ -1324,30 +1324,40 @@ func! Test_edit_rightleft()
 endfunc
 
 func Test_edit_complete_very_long_name()
-  if !has('unix') || has('mac')
-    " Long directory names only work on Unix.
-    return
-  endif
   " Try to get the Vim window position before setting 'columns'.
   let winposx = getwinposx()
   let winposy = getwinposy()
   let save_columns = &columns
   set columns=2000
-  call assert_equal(2000, &columns)
+  if has('unix')
+    " On windows (Appveyor) '&columns' would be limited to up to 123.
+    call assert_equal(2000, &columns)
+  endif
   set noswapfile
 
   let dirname = getcwd() . "/Xdir"
-  let longdirname = dirname . repeat('/' . repeat('d', 255), 4)
-  let longfilename = longdirname . '/' . repeat('a', 255)
-  call mkdir(longdirname, 'p')
-  call writefile(['Totum', 'Table'], longfilename)
-  new
-  exe "next Xfile " . longfilename
-  exe "normal iT\<C-N>"
+  if has('win32')
+    let path_max = 260
+  else
+    " 'PATH_MAX' on linux is 4096 and on other *nix seems to be 1024.
+    let path_max = 1024
+  endif
```

Comment:
A discussion about PATH_MAX was already done at https://groups.google.com/forum/#!topic/vim_dev/nHMRy5P5K6M

At that time, our conclusion was that the limitation of 1024 bytes for macOS would make the test pointless.  As the proposed limitation for Windows is 200, the conclusion remains upheld from that point of view.

So, can you tell us why you think the test is still needed for macOS and Window (and maybe other Unix variants)?  There might be other reasons we are missing to justify the proposed modification.